File: Python/Warmup-2/string_match.py
"""
Dada 2 Strings ("a" e "b"), retorne o número de substrings de tamanho 2 que ambas
possuem coincidentemente e no mesmo índice de local.
Ex.:(('xxcaazz', 'xxbaaz') → 3; ('abc', 'abc') → 2; ('abc', 'axc') → 0).
"""


def string_match(a, b):
    cont = 0
    for i in range(min(len(a), len(b))-1):
        if a[i:i+2] == b[i:i+2]:
            cont += 1
    return cont


print(string_match('xxcaazz', 'xxbaaz'))

File: Java/Warmup-2/stringMatch/StringMatch.java
package stringMatch;

public class StringMatch {
    //Retorne o número de subtrings de tamanho 2 que sejam comuns nas duas Strings e estejam nas mesmas posições.
    //(EX.: ("abcx", "abdcx") - > 1).

    public int stringMatch(String a, String b) {
        int size, cont = 0;
        size = Math.min(a.length(), b.length());
        for (int i = 0; i < size - 1; i++) {
            if ((a.charAt(i) + a.charAt(i+1)) == (b.charAt(i) + b.charAt(i+1))) cont++;
        }
        return cont;
    }
}
